### PR TITLE
feat: Support Prometheus Operator in Helm chart

### LINKED ISSUE #7: Support Prometheus Operator in Helm chart
> The chart should (optionally) create a ServiceMonitor object so it can seamlessly be integrated into a Prometheus Operator deployment.
> 
> Some examples can be found here:
> - https://github.com/helm/charts/blob/master/stable/prometheus-postgres-exporter/templates/servicemonitor.yaml
> - https://github.com/helm/charts/blob/master/stable/prometheus-redis-exporter/templates/servicemonitor.yaml
> - https://github.com/helm/charts/blob/master/stable/prometheus-rabbitmq-exporter/templates/servicemonitor.yaml

Originally requested by @DaveOHenry in https://github.com/tomkerkhove/promitor/issues/885

### PR DESCRIPTION
**Pull request by DaveOHenry**
Fixes #7 

This PR adds support for ServiceMonitor and PrometheusRule objects and some useful configuration options for both features.

### CODE REVIEW

File: promitor-agent-scraper/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
-version: 2.1.0
+version: 2.2.0
```
Comment (tomkerkhove) [inline promitor-agent-scraper/Chart.yaml:2]: Please revert, I'll ship it asap seperately

File: promitor-agent-scraper/templates/prometheusrule.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{{- if .Values.prometheusRule.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ template "promitor-agent-scraper" . }}
+{{- with .Values.prometheusRule.namespace }}
+  namespace: {{ . }}
+{{- end }}
+  labels:
```
Comment (tomkerkhove) [inline promitor-agent-scraper/templates/prometheusrule.yaml:9]: Can you align with the other resources here please?

https://github.com/promitor/charts/blob/main/promitor-agent-scraper/templates/deployment.yaml#L7

File: promitor-agent-scraper/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{{- if and ( .Values.metricSinks.prometheusScrapingEndpoint.serviceMonitor.enabled ) (.Values.metricSinks.prometheusScrapingEndpoint.enabled ) }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+{{- if .Values.metricSinks.prometheusScrapingEndpoint.serviceMonitor.labels }}
+  labels:
+{{ toYaml .Values.metricSinks.prometheusScrapingEndpoint.serviceMonitor.labels | indent 4 }}
```
Comment (tomkerkhove) [inline promitor-agent-scraper/templates/servicemonitor.yaml:7]: Same here, can we at least add our default labels?
https://github.com/promitor/charts/blob/main/promitor-agent-scraper/templates/deployment.yaml#L7

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Looks good, thanks!

Just a few remarks

File: promitor-agent-scraper/templates/prometheusrule.yaml (hunk 2 above)
Comment (DaveOHenry) [reply promitor-agent-scraper/templates/prometheusrule.yaml:9]: Sure. That makes sense.

File: promitor-agent-scraper/templates/servicemonitor.yaml (hunk 3 above)
Comment (DaveOHenry) [reply promitor-agent-scraper/templates/servicemonitor.yaml:7]: Same here 👍

**Review by DaveOHenry — COMMENTED** [review]
Ran some tests with a "kube-prometheus-stack" deployment. Looks good so far. I have to collect all the resources now, define the metrics and add alerting rules 😃

**Review by tomkerkhove — APPROVED** [review]
LGTM, thanks a ton - This is super helpful!

**Pull request merged**